tabbycat/tournaments/views.py

```diff
@@ -130,9 +130,23 @@ def get_context_data(self, **kwargs):
         kwargs['increment_ok'] = kwargs['num_unconfirmed'] == 0
         kwargs['emails_sent'] = BulkNotification.objects.filter(
             tournament=self.tournament, round=self.round, event=BulkNotification.EventType.POINTS).exists()
+        kwargs['silent'] = self.round.silent
         return super().get_context_data(**kwargs)
 
 
+class CompleteRoundToggleSilentView(AdministratorMixin, RoundMixin, PostOnlyRedirectView):
+    def post(self, request, *args, **kwargs):
+        self.round.silent = not self.round.silent
+        self.round.save()
+
+        if self.round.silent:
+            messages.success(request, _("%(round)s has been marked as silent.") % {'round': self.round.name})
+        else:
+            messages.success(request, _("%(round)s has been unmarked as silent.") % {'round': self.round.name})
+
+        return redirect_round('tournament-complete-round-check', self.round)
```

Comment:
Can set `round_redirect_pattern_name = 'tournament-complete-round-check'` as a class attribute, then return `super` rather than the redirect directly.